git_machete/github.py

```diff
@@ -104,282 +163,296 @@ def get_token_from_gh() -> Optional[GithubTokenAndTokenProvider]:
-    return str(user['login'])  # str() to satisfy mypy
-
-
-def is_github_remote_url(domain: str, url: str) -> bool:
-    return any((re.match(pattern, url) for pattern in github_remote_url_patterns(domain)))
+            return None
 
+    def derive_pull_requests(self) -> List[GitHubPullRequest]:
+        # As of Dec 2022, GitHub API never returns more than 100 PRs, even if per_page>100.
+        prs = self.__fire_github_api_request(method='GET',
+                                             path=f'/repos/{self.__organization}/{self.__repository}/pulls?per_page=100')
+        return list(map(GitHubPullRequest.from_json, prs))
 
-def get_parsed_github_remote_url(domain: str, url: str, remote: str) -> Optional[RemoteAndOrganizationAndRepository]:
-    for pattern in github_remote_url_patterns(domain):
-        match = re.match(pattern, url)
-        if match:
-            org = match.group(1)
-            repo = match.group(2)
-            return RemoteAndOrganizationAndRepository(remote=remote,
-                                                      organization=org,
-                                                      repository=repo if repo[-4:] != '.git' else repo[:-4])
-    return None
-
+    def derive_current_user_login(self) -> Optional[str]:
+        if not self.__token:
+            return None
+        user = self.__fire_github_api_request(method='GET',
+                                              path='/user')
+        return str(user['login'])  # str() to satisfy mypy
+
+    def get_pull_request_by_number_or_none(self,
+                                           number: int
+                                           ) -> Optional[GitHubPullRequest]:
+        try:
+            path = f'/repos/{self.__organization}/{self.__repository}/pulls/{number}'
+            pr_json: Dict[str, Any] = self.__fire_github_api_request(method='GET',
+                                                                     path=path)
+            return GitHubPullRequest.from_json(pr_json)
+        except MacheteException:
+            return None
 
-def get_pull_request_by_number_or_none(domain: str, number: int, org: str, repo: str) -> Optional[GitHubPullRequest]:
-    token: Optional[str] = __get_github_token(domain)
-    try:
-        pr_json: Dict[str, Any] = __fire_github_api_request(domain, 'GET', f'/repos/{org}/{repo}/pulls/{number}', token)
-        return __parse_pr_json(pr_json)
-    except MacheteException:
-        return None
+    def get_repo_and_org_names_by_id(self,
+                                     repo_id: str
+                                     ) -> str:
+        repo = self.__fire_github_api_request(path='GET',
+                                              method=f'/repositories/{repo_id}')
+        return str(repo['full_name'])
 
 
-def checkout_pr_refs(git: GitContext, remote: str, pr_number: int, branch: LocalBranchShortName) -> None:
+def checkout_pr_refs(git: GitContext,
```

Comment:
Nit: I'd extract a method like `get_remote_ref_for_github_pr` which would return `'pull/{pr_number}/head:{branch}'`, but move the method `checkout_pr_refs` to `MacheteClient`

Reply:
@PawelLipski I will move the method to the class but i don't think that extracting a new method is necessary here since all the paths for each `MacheteClient` method are coded inside the method so i would stick to it here as well.